packages/@lwc/template-compiler/src/codegen/index.ts

```diff
@@ -71,7 +72,12 @@ function transform(codeGen: CodeGen): t.Expression {
         const databag = elementDataBag(element, slotParentName);
         let res: t.Expression;
 
-        if (codeGen.staticNodes.has(element) && isElement(element)) {
+        if (
+            codeGen.staticNodes.has(element) &&
+            isElement(element) &&
+            // Tags like <th> and <td> are okay as static fragments, but only if they're not at the top level
+            !isNonTopLevelTag(element)
+        ) {
```

Comment:
I actually noticed this was a problem because in one of the snapshots we had a fragment like:

```html
<table><tbody><tr><td></td></tr></tbody></table>
```

We should totally allow these kinds of fragments. There's no problem with them. The only problem is if elements like `tbody` and `td` appear at the top level of a fragment.